### DISCUSSION TITLE
Cart note not updating after customer visits checkout

### DISCUSSION THREAD

**Discussion by justenh** [Shopify/storefront-api-feedback#217] [Help]
We recently launched our custom storefront and encountered an odd behavior with the StoreFront API that we are struggling to fix, and has pretty large implications for how customers communicate shipping and gift information to us.

Orders do not receive the latest Cart note after customer clicks checkout. Steps to recreate...

1. Customer adds a note to the cart ex('Please ship on 12/20/2023')
2. Customer attempts to checkout (goes to Shopify checkout experience)
3. Customer leaves Shopify checkout and goes back to cart
4. Customer updates note ex ('Please ship immediately')
5. Customer goes back to checkout and completes the checkout
6. **The note does not get updated, and the order is recorded with the original cart note**

I can confirm the note is accurately recorded to the cart, because we load it and show it on the Cart page. It's only in the final order that the note does not get updated. Any help would be greatly appreciated! 

Our website uses the Storefront API with PHP and does not use the Hydrogen framework.

**Comment by johncraigcole** [comment]
Hey @justenh 👋 . After updating the note, does the buyer follow the `checkoutUrl` of the Storefront API cart?

I've done some testing and I'm not able to reproduce the flow as you've described above. I would definitely reach out to support if the problem continues.

**Reply by justenh** [reply]
@johncraigcole Thank you for looking into this issue. 

That's correct. The user ultimately follows the `checkoutUrl` provided by the Storefront API cart. 

The cart is updated with the note via a full page refresh that sends it to the Storefront API. The customer is then brought back to the cart. Once the customer clicks a "Check Out" button the site is redirected to the URL you mentioned provided by the Cart API.

We know the cart API is working correctly because during the full page refresh, the customer's new note is correctly read from the Cart object returned by Shopify and displayed back to them. Additionally, the URL for the checkout is only retrieved once they click the checkout button by doing a redirect to an internal page that then reads the Shopping Cart data provided by the Storefront API and redirects accordingly. 

Your colleague Gray (Shopify), shopigray, on Discord, mentioned being able to recreate the issue when I first reported it there.

Please let me know if there's any additional information I can provide. I'd also be happy to set up a DM on Discord where I can walk you through the issue first hand.

Thanks again!

**Reply by johncraigcole** [reply]
Can you share the discord thread?

**Reply by johncraigcole** [reply]
Nevermind found it.

**Reply by johncraigcole** [reply]
I realized in my testing of the issue I had a note specified on creation which seems to have a different behavior. However, when creating cart with no note, and then adding a note after with `cartNoteUpdate` I get the same behavior you've documented.

We have an internal issue created with the appropriate team. I don't have a timeline on a fix but will follow up here when it ships to production.

**Reply by justenh** [reply]
@johncraigcole Thanks again for your help. I Wanted to check in and see if this fix has had any progress? We continue to have a number customer's who are affected by this. It turns into a pretty sticky customer support problem because our customers are often attempting to apply notes to how their product should be shipped.

Thanks again for your help,
Justen

**Reply by johncraigcole** [reply]
Hey @justenh, there was a fix shipped to production for this back in mid to late December. Judging by your comment, it sounds like you're still seeing the issue. Are you on a Plus plan that is still on checkout.liquid? The fix was pushed to the checkout extensibility platform, which should mean that everyone not on Plus, and Plus merchants that have migrated to checkout extensibility should have gotten the fix.

I re-tested using the same workflow with my test store and no longer see the issue.

**Reply by justenh** [reply]
@johncraigcole Thanks for the update. I was just going by the open status of this issue  and haven't confirmed it with our team. I'll get a confirmation from them and report back. Appreciate your help!

**Comment by justenh** [comment]
Thanks for the update! That makes sense. We create our cart very early on
in the process  which means almost all of our carts are created without a
note first.

I'll continue to monitor this thread for the fix. Thank you for your
support.

On Tue, Oct 10, 2023 at 6:02 PM JohnCCole ***@***.***> wrote:

> I realized in my testing of the issue I had a note specified on creation
> which seems to have a different behavior. However, when creating cart with
> no note, and then adding a note after with cartNoteUpdate I get the same
> behavior you've documented.
>
> We have an internal issue created with the appropriate team. I don't have
> a timeline on a fix but will follow up here when it ships to production.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/Shopify/storefront-api-feedback/discussions/217#discussioncomment-7246698>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AA5QLOZSQPTVZIJF6UHZXITX6XHXFAVCNFSM6AAAAAA5WCNR4CVHI2DSMVQWIX3LMV43SRDJONRXK43TNFXW4Q3PNVWWK3TUHM3TENBWGY4TQ>
> .
> You are receiving this because you were mentioned.Message ID:
> <Shopify/storefront-api-feedback/repo-discussions/217/comments/7246698@
> github.com>
>


-- 
From Somebody Else
Justen Holter

- 305.343.1611
- ***@***.***
- www.fromsomebodyelse.com
